### PR TITLE
(CDAP-15790) Remove support for HBase 0.96 and 0.98

### PR DESCRIPTION
**Pull request by chtyim**
- Change to use HBase Table instead of HTable
- Fix resources leakage in unit-tests

### CODE REVIEW

**Comment by chtyim** [comment]
Build passed https://builds.cask.co/browse/CDAP-DUT7049

File: cdap-hbase-compat-base/src/main/java/io/cdap/cdap/hbase/wd/DistributedScanner.java [hunk 1]
```diff
@@ -94,7 +93,7 @@ public Result next() throws IOException {
         break;
       }
     }
-    return resultSets.toArray(new Result[resultSets.size()]);
+    return resultSets.toArray(new Result[0]);
```
Comment (yaojiefeng) [inline cdap-hbase-compat-base/src/main/java/io/cdap/cdap/hbase/wd/DistributedScanner.java:96]: Can we just use `toArray()` here? I think new Object[0] is identical to it.

File: cdap-hbase-compat-base/pom.xml [hunk 2]
```diff
@@ -50,6 +50,13 @@
       <artifactId>tephra-core</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-common</artifactId>
+      <version>${hadoop26.version}</version>
```
Comment (yaojiefeng) [inline cdap-hbase-compat-base/pom.xml:56]: This change is because hbase 1.0 uses hadoop 2.6 right? I am wondering how it works for higher version of hbase compat. If a compact depends on higher version hadoop, will it still work?

File: cdap-hbase-compat-base/src/main/java/io/cdap/cdap/data2/util/hbase/HBaseTableUtil.java [hunk 3]
```diff
@@ -722,4 +736,51 @@ public int getPriority() {
...
+        oldCount = connectionCount.get();
+        if (oldCount == 0) {
+          return null;
+        }
+      } while (!connectionCount.compareAndSet(oldCount, oldCount + 1));
+      return connection;
+    }
+
+    void releaseConnection() throws IOException {
+      if (connectionCount.decrementAndGet() == 0) {
```
Comment (yaojiefeng) [inline cdap-hbase-compat-base/src/main/java/io/cdap/cdap/data2/util/hbase/HBaseTableUtil.java]: Should we log something if the count is greater than 0? Since this method is called in the `close()` method, but if the count > 0, this connection is actually not closed without any exception indicated.

File: cdap-data-fabric-tests/pom.xml [hunk 4]
```diff
@@ -124,6 +124,14 @@
 
   <build>
     <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <configuration>
+          <argLine>-Xmx6000m -Djava.awt.headless=true -XX:+UseG1GC -XX:OnOutOfMemoryError="kill -9 %p" -XX:+HeapDumpOnOutOfMemoryError</argLine>
```
Comment (yaojiefeng) [inline cdap-data-fabric-tests/pom.xml:131]: What is the purpose of this change?

**Review by yaojiefeng — APPROVED** [review]
LGTM, just few minor comments

File: cdap-hbase-compat-base/src/main/java/io/cdap/cdap/hbase/wd/DistributedScanner.java (hunk 1 above)
Comment (chtyim) [reply cdap-hbase-compat-base/src/main/java/io/cdap/cdap/hbase/wd/DistributedScanner.java:96]: `toArray()` returns `Object[]` instead of `Result[]`.

File: cdap-hbase-compat-base/pom.xml (hunk 2 above)
Comment (chtyim) [reply cdap-hbase-compat-base/pom.xml:56]: It depends. When we have higher hbase compat and if doesn't work, we'll address that.

File: cdap-hbase-compat-base/src/main/java/io/cdap/cdap/data2/util/hbase/HBaseTableUtil.java (hunk 3 above)
Comment (chtyim) [reply cdap-hbase-compat-base/src/main/java/io/cdap/cdap/data2/util/hbase/HBaseTableUtil.java]: Added a trace log.

File: cdap-data-fabric-tests/pom.xml (hunk 4 above)
Comment (chtyim) [reply cdap-data-fabric-tests/pom.xml:131]: To have more memory and reduce the parallel runs for the data-fabric-tests module. This is because HBase 1.0 uses more memory than HBase 0.98
